File: apps/web/shared/types/friend-link.ts
export type FriendLinkCategory = 'official' | 'galgame' | 'others'

export type FriendLinkStatus = 'normal' | 'essential' | 'down'

export interface FriendLink {
  id: number
  category: FriendLinkCategory
  name: string
  link: string
  description: string
  banner: string
  banner_image_hash: string
  banner_url: string
  status: FriendLinkStatus
  sort_order: number
  created: string
  updated: string
}

export interface GroupedFriendLinks {
  official: FriendLink[]
  galgame: FriendLink[]
  others: FriendLink[]
}

export interface FriendLinkInput {
  id?: number
  category: FriendLinkCategory
  name: string
  link: string
  description: string
  banner: string
  banner_image_hash: string
  status: FriendLinkStatus
}

File: apps/api/internal/friendlink/model/friendlink.go
package model

import "time"

type FriendLink struct {
	ID              int       `gorm:"primaryKey;autoIncrement" json:"id"`
	Category        string    `gorm:"not null" json:"category"`
	Name            string    `gorm:"not null" json:"name"`
	Link            string    `gorm:"not null" json:"link"`
	Description     string    `gorm:"type:text;default:''" json:"description"`
	Banner          string    `gorm:"default:''" json:"banner"`
	BannerImageHash string    `gorm:"column:banner_image_hash;default:''" json:"banner_image_hash"`
	BannerURL       string    `gorm:"-" json:"banner_url"`
	Status          string    `gorm:"default:'normal'" json:"status"`
	SortOrder       int       `gorm:"column:sort_order;not null;default:0" json:"sort_order"`
	CreatedAt       time.Time `gorm:"column:created" json:"created"`
	UpdatedAt       time.Time `gorm:"column:updated" json:"updated"`
}

func (FriendLink) TableName() string { return "friend_link" }
